src/tests/msw-handlers/delete-contact.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { ErrorSoapResponse, SoapResponse } from '@zextras/carbonio-shell-ui';
+import { ResponseResolver, rest, RestContext, RestRequest } from 'msw';
+
+import { getSetupServer } from '../../carbonio-ui-commons/test/jest-setup';
+import { NAMESPACES } from '../../constants/api';
+import {
+	CONTACT_ACTION_OPERATION,
+	ContactActionRequest,
+	ContactActionResponse
+} from '../../network/api/contact-action';
+import { buildSoapResponse } from '../utils';
+
+type DeleteContactHandler = ResponseResolver<
+	RestRequest<{ Body: { ContactActionRequest: ContactActionRequest } }>,
+	RestContext,
+	SoapResponse<ContactActionResponse>
+>;
+export const registerDeleteContactHandler = (
+	id: string,
+	error?: string
+): jest.Mock<ReturnType<DeleteContactHandler>, Parameters<DeleteContactHandler>> => {
+	const handler = jest.fn<ReturnType<DeleteContactHandler>, Parameters<DeleteContactHandler>>(
+		(req, res, ctx) => {
+			if (error) {
+				return res(
+					ctx.json<ErrorSoapResponse>({
+						Header: {
+							context: {}
+						},
+						Body: {
+							Fault: {
+								Reason: { Text: error },
+								Detail: {
+									Error: { Code: '', Detail: error }
+								}
+							}
+						}
+					})
+				);
```

Comment:
Check this util to reduce the "useless" lines
https://github.com/zextras/carbonio-contacts-ui/blob/8915b79e0197380a1b63b2af1d2897d556412d3e/src/tests/utils.ts#L43-L61